rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -38,29 +39,51 @@ func resourceRancher2MachineConfigV2Create(d *schema.ResourceData, meta interfac
 
 	d.SetId(newObj.ID)
 	d.Set("kind", newObj.TypeMeta.Kind)
-	stateConf := &resource.StateChangeConf{
-		Pending:    []string{},
-		Target:     []string{"active"},
-		Refresh:    machineConfigV2StateRefreshFunc(meta, newObj.ID, newObj.TypeMeta.Kind),
-		Timeout:    d.Timeout(schema.TimeoutCreate),
-		Delay:      1 * time.Second,
-		MinTimeout: 3 * time.Second,
-	}
-	_, waitErr := stateConf.WaitForState()
-	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
+
+	err = waitForMachineConfigV2State(meta.(*Config), d, meta, d.Timeout(schema.TimeoutCreate))
```

Comment:
If I understand correctly, this create timeout is 20 or 30 minutes. That is appropriate for clusters, but this timeout should likely be shorter: 30-60 seconds.

Additionally, the name of this function doesn't explain what it does. It implies that there is some state the machine config is going to transition to.

Reply:
I'm using the machine config default create timeout https://github.com/annablender/terraform-provider-rancher2/blob/ca13e143dedd568ef010339bc611041aaedd04aa/rancher2/resource_rancher2_machine_config_v2.go#L21, which is 10 mins. Is changing it to 1 min ok?

The func name is cohesive with the wait func for the cluster_v2 resource https://github.com/annablender/terraform-provider-rancher2/blob/ca13e143dedd568ef010339bc611041aaedd04aa/rancher2/resource_rancher2_cluster_v2.go#L265. If I change one, would I have to change both? I guess, maybe `waitForValidMachineConfigV2State` would be more specific.

Reply:
I think 10 minutes is too long. There is nothing that has to happen when the machine config is created besides the RBAC getting applied. I think 1 to 3 minutes is plenty.

The cluster function takes in a `state` parameter, so the name makes sense there. The machine config one does not. Also, you are passing the same parameter twice in your implementation. Is it possible to only pass `meta` in once?

Reply:
Right, and with this implementation the machine config restarts RBAC once within 20 secs and terraform succeeds in getting the object. We don't want to underestimate the timeout though, so 3 mins sounds fair. For the name, how about `waitForMachineConfigV2` or `getMachineConfigV2?` Yeah, thanks for catching that. I only have to pass in `meta` and can grab `meta.(*Config)` inside the func.